pkg/activator/net/revision_backends.go

```diff
@@ -212,31 +219,38 @@ func (rw *revisionWatcher) checkDests(dests []string) {
 			// we have a healthy clusterIP but revision is not ready. We must have scaled down.
 			rw.clusterIPHealthy = false
 
-			// Send update that were now inactive
+			// Send update that we are now inactive
 			rw.sendUpdate(&RevisionDestsUpdate{Rev: rw.rev})
 		}
 
 		// We know this revision cannot be healthy so short circuit
 		return
 	}
 
-	if rw.clusterIPHealthy {
-		// cluster IP is healthy and we havent scaled down, short circuit
-		return
-	}
-
 	// First check the clusterIP
 	svc, err := rw.getK8sPrivateService()
 	if err != nil {
 		rw.logger.Errorw("Failed to lookup private service for revision", zap.Error(err))
 		return
 	}
 
+	if rw.clusterIPHealthy {
+		dest, err := rw.getDest(svc)
+		if err != nil {
+			rw.logger.Errorw("Failed to determine service destination", zap.Error(err))
+			return
+		}
+		// cluster IP is healthy and we havent scaled down, short circuit
+		rw.logger.Debugf("ClusterIP %s already probed (backends: %d)", dest, len(dests))
+		rw.sendUpdate(&RevisionDestsUpdate{Rev: rw.rev, ClusterIPDest: dest, ReadyAddressCount: len(dests)})
+		return
+	}
+
 	// If clusterIP is healthy send this update and we are done
```

Comment:
This comment doesn't make sense?  Can you please remove or update?

Reply:
I think it still makes sense.  We probe the ClusterIP and send an update if it is healthy (the else clause)